src/transformers/integrations/peft.py

```diff
@@ -357,6 +373,15 @@ def get_adapter_state_dict(self, adapter_name: Optional[str] = None) -> dict:
         if adapter_name is None:
             adapter_name = self.active_adapter()
 
+        if isinstance(adapter_name, list):
```

Comment:
Is it possible for this method to be called with a list of str? Wouldn't this require the user to explicitly pass that argument? If not, I think this check is not necessary. If it is a valid argument, then the type annotation should also be adjusted.

Reply:
Good point, I don't think we should allow users to call this method with a list of str - I refactored a bit the logic of `def active_adapter` to extend it for multi-adapter inference. let me know what do you think